src/main/java/Duke/Task/Deadline.java

```diff
@@ -0,0 +1,38 @@
+package Duke.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represent a deadline object with a deadline timing.
+ */
+public class Deadline extends Task {
+    
+    protected LocalDateTime by;
+
+    /**
+     * Construct a deadline object.
+     * @param description title of the deadline.
+     * @param by deadline timing.
+     * @param isDone status.
+     */
+    public Deadline(String description, LocalDateTime by, boolean isDone) {
+        super(description, isDone);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM-dd-yyyy HH:mm");
+        return "[D]" + super.toString() + "(by: " + by.format(format) + ")";
+    }
+
+    /**
+     * Represents the string written into data.txt.
```

Comment:
In Javadoc comments, there should be an empty line between description and return section.